plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFEInputSpent.sol

```diff
@@ -0,0 +1,87 @@
+    using UtxoPosLib for UtxoPosLib.UtxoPos;
+    using PaymentInFlightExitModelUtils for PaymentExitDataModel.InFlightExit;
+
+    // TODO: Use BondSize lib.
+    uint256 public constant PIGGYBACK_BOND = 31415926535 wei;
+
+    struct Controller {
+        PlasmaFramework framework;
+        PaymentSpendingConditionRegistry spendingConditionRegistry;
+        uint256 supportedTxType;
+        EthVault ethVault;
+    }
+
+    event InFlightExitInputBlocked(
+        address indexed challenger,
+        bytes32 txHash,
+        uint8 inputIndex
+    );
+
+    function run(
+        Controller memory self,
+        PaymentExitDataModel.InFlightExitMap storage inFlightExitMap,
+        PaymentInFlightExitRouterArgs.ChallengeInputSpentArgs memory args
+    )
+        public
+    {
+        uint192 exitId = ExitId.getInFlightExitId(args.inFlightTx);
+        PaymentExitDataModel.InFlightExit storage ife = inFlightExitMap.exits[exitId];
+
+        require(ife.exitStartTimestamp != 0, "In-flight exit doesn't exist");
+
+        require(ife.isInputPiggybacked(args.inFlightTxInputIndex), "The indexed input has not been piggybacked");
+
+        require(
+            keccak256(args.inFlightTx) != keccak256(args.spendingTx),
+            "The spending transaction is the same as the in-flight transaction"
+        );
+
+        PaymentTransactionModel.Transaction memory spendingTx = PaymentTransactionModel.decode(args.spendingTx);
+        require(args.spendingTxInputIndex < spendingTx.inputs.length, "Incorrect spending transaction input index");
+
+        bytes32 spendingInput = spendingTx.inputs[args.spendingTxInputIndex];
+        bytes32 ifeInput = PaymentTransactionModel.decode(args.inFlightTx).inputs[args.inFlightTxInputIndex];
+        require(ifeInput == spendingInput, "Spent input is not the same as piggybacked input");
+
+        IPaymentSpendingCondition condition = self.spendingConditionRegistry.spendingConditions(
+            args.spendingTxInputOutputType, spendingTx.txType
+        );
+        require(address(condition) != address(0), "Spending condition contract not found");
+
+        // FIXME: move to the finalized interface as https://github.com/omisego/plasma-contracts/issues/214
+        // Also, the tests should verify the args correctness
+        bool isSpentByInFlightTx = condition.verify(
+            bytes32(''), // tmp solution, we don't need outputGuard anymore for the interface of :point-up: GH-214
+            uint256(0), // should not be used
+            ife.inputs[args.inFlightTxInputIndex].outputId,
+            args.spendingTx,
+            args.spendingTxInputIndex,
+            args.spendingTxWitness
+        );
+        require(isSpentByInFlightTx, "Spent input spending condition is not met");
+
+        // Remove the input from the piggyback map
+        ife.clearInputPiggybacked(args.inFlightTxInputIndex);
+
+        // Pay out the bond.
+        self.ethVault.withdraw(msg.sender, PIGGYBACK_BOND);
```

Comment:
so far the current implementation would actually keep the bond fund on the ExitGame contract directly instead of using vault since the "value" in transaction would by default goes to ExitGame when start exits...(and we do not explicit move funds to vault so far)

Reply:
Oh, I didn't notice that 🤦‍♂ 
So bonds don't ever go into the vault. Is that on purpose? The amounts are low anyway, so I guess it's simpler?

Reply:
it's not on purpose, we can change that if that seems to simply things?
It is just currently we didn't have extra code in start exit to move fund from ExitGame contract to vault, and I don't know which is better to be honest.

Reply:
Ok. For now I'll change this to be consistent with other bonds (i.e. keep it in the contract),
and create an issue to decide if we want to move it to the vault